packages/rich-text/src/plugins/Tracking/createTrackingPlugin.ts

```diff
@@ -111,5 +111,55 @@ export const createTrackingPlugin = (onAction: RichTextTrackingActionHandler): R
 
       return editor;
     },
+    then: (editor) => ({
+      editor: {
+        insertData: {
+          format: 'text/html',
+          getFragment: ({ data }) => {
+            let source;
+
+            if (data.includes('docs-internal-guid')) {
+              source = 'Google Docs';
+            }
+
+            if (data.includes('<google-sheets-html-origin>') || data.includes('data-sheets-')) {
+              source = 'Google Spreadsheets';
+            }
+
+            // TODO: MS Word Online doesn't give us specific tags, we might need to have a closer look at its tracking result since we are using generic values to identify it
+            if (
+              data.includes('Arial_MSFontService') &&
+              (data.includes('class="TextRun') || data.includes('class="OutlineElement'))
+            ) {
+              source = 'Microsoft Word Online';
+            }
+
+            if (data.includes('<meta name=Generator content="Microsoft Excel')) {
+              source = 'Microsoft Excel';
+            }
+
+            if (data.includes('<meta name=Generator content="Microsoft Word')) {
+              source = 'Microsoft Word';
+            }
+
+            if (data.includes('<meta name="Generator" content="Cocoa HTML Writer">')) {
+              source = 'Apple Notes';
+            }
+
+            if (data.includes('Slack-Lato, Slack-Fractions')) {
+              source = 'Slack';
+            }
+
+            if (source) {
+              editor.tracking.onShortcutAction('paste', {
```

Comment:
We have other places where we fire `paste` events e.g. in the table plugin, wouldn't that mean it would fire twice? making the data less real. Can we combine these calls somehow?